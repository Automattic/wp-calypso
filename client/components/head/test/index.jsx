/**
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import React from 'react';
import Head from '../';

describe( 'Head', () => {
	test( 'should render default title', () => {
		const wrapper = shallow( <Head /> );

		expect( wrapper.find( 'title' ).text() ).toBe( 'WordPress.com' );
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render custom title', () => {
		const title = 'Arbitrary Custom Title';
		const wrapper = shallow( <Head title={ title } /> );

		expect( wrapper.find( 'title' ).text() ).toBe( title );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
