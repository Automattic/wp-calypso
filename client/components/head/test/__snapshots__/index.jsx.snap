// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head should render custom title 1`] = `
<head>
  <title>
    Arbitrary Custom Title
  </title>
  <meta
    charSet="utf-8"
  />
  <meta
    content="IE=Edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <meta
    content="telephone=no"
    name="format-detection"
  />
  <meta
    content="yes"
    name="mobile-web-app-capable"
  />
  <meta
    content="yes"
    name="apple-mobile-web-app-capable"
  />
  <meta
    content="#055d9c"
    name="theme-color"
  />
  <meta
    content="origin"
    name="referrer"
  />
  <link
    as="document"
    href="https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0"
    rel="prefetch"
  />
  <Favicons
    environmentFaviconURL="//s1.wp.com/i/favicon.ico"
  />
  <link
    href="http://gmpg.org/xfn/11"
    rel="profile"
  />
  <link
    href="/calypso/manifest.json"
    rel="manifest"
  />
  <link
    as="style"
    href="https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese&display=swap"
    rel="preload"
  />
  <noscript>
    <link
      href="https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese&display=swap"
      rel="stylesheet"
    />
  </noscript>
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
			(function() {
				var m = document.createElement( \\"link\\" );
				m.rel = \\"stylesheet\\";
				m.href = \\"https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese&display=swap\\";
				document.head.insertBefore( m, document.head.childNodes[ document.head.childNodes.length - 1 ].nextSibling );
			})()
			",
      }
    }
    type="text/javascript"
  />
</head>
`;

exports[`Head should render default title 1`] = `
<head>
  <title>
    WordPress.com
  </title>
  <meta
    charSet="utf-8"
  />
  <meta
    content="IE=Edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <meta
    content="telephone=no"
    name="format-detection"
  />
  <meta
    content="yes"
    name="mobile-web-app-capable"
  />
  <meta
    content="yes"
    name="apple-mobile-web-app-capable"
  />
  <meta
    content="#055d9c"
    name="theme-color"
  />
  <meta
    content="origin"
    name="referrer"
  />
  <link
    as="document"
    href="https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0"
    rel="prefetch"
  />
  <Favicons
    environmentFaviconURL="//s1.wp.com/i/favicon.ico"
  />
  <link
    href="http://gmpg.org/xfn/11"
    rel="profile"
  />
  <link
    href="/calypso/manifest.json"
    rel="manifest"
  />
  <link
    as="style"
    href="https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese&display=swap"
    rel="preload"
  />
  <noscript>
    <link
      href="https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese&display=swap"
      rel="stylesheet"
    />
  </noscript>
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
			(function() {
				var m = document.createElement( \\"link\\" );
				m.rel = \\"stylesheet\\";
				m.href = \\"https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese&display=swap\\";
				document.head.insertBefore( m, document.head.childNodes[ document.head.childNodes.length - 1 ].nextSibling );
			})()
			",
      }
    }
    type="text/javascript"
  />
</head>
`;
